src/MongoDB.Analyzer/Core/ExpressionProcessor.cs

```diff
@@ -0,0 +1,468 @@
+﻿// Copyright 2021-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using MongoDB.Analyzer.Core.HelperResources;
+using TypeInfo = Microsoft.CodeAnalysis.TypeInfo;
+
+namespace MongoDB.Analyzer.Core;
+
+internal static class ExpressionProcessor
+{
+    public record RewriteContext(
+        AnalysisType AnalysisType,
+        SyntaxNode Expression,
+        SyntaxNode RootNode,
+        SemanticModel SemanticModel,
+        TypesProcessor TypesProcessor,
+        ConstantsMapper ConstantsMapper)
+    {
+        public static RewriteContext Builders(SyntaxNode Expression, SyntaxNode RootNode, SemanticModel SemanticModel, TypesProcessor TypesProcessor) =>
+            new(AnalysisType.Builders, Expression, RootNode, SemanticModel, TypesProcessor, new());
+
+        public static RewriteContext Linq(SyntaxNode Expression, SyntaxNode RootNode, SemanticModel SemanticModel, TypesProcessor TypesProcessor) =>
+            new(AnalysisType.Linq, Expression, RootNode, SemanticModel, TypesProcessor, new());
+    }
+
+    private enum RewriteAction
```

Comment:
Usually it's nice idea to have default enum value to some Unknown value. To avoid unintended default value in a variable.

Reply:
Agreed, done.